packages/search-input/src/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,377 @@
+            closeMenu();
+            inputRef.current?.focus();
+            break;
+          }
+
+          case keyMap.ArrowDown: {
+            if (withTypeAhead) {
+              inputRef.current?.focus();
+              e.preventDefault(); // Stop page scroll
+              updateHighlight('next');
+            }
+            break;
+          }
+
+          case keyMap.ArrowUp: {
+            if (withTypeAhead) {
+              inputRef.current?.focus();
+              e.preventDefault(); // Stop page scroll
+              updateHighlight('prev');
+            }
+            break;
+          }
+        }
+      }
+    };
+
+    const handleBlur: FocusEventHandler = e => {
+      if (
+        !isElementInComponent(e.target) &&
+        !isElementInComponent(document.activeElement)
+      ) {
+        closeMenu();
+      }
+    };
+
+    const handleSubmit: FormEventHandler<HTMLFormElement> = e => {
+      if (withTypeAhead) {
+        e.preventDefault();
+      } else {
+        onSubmitProp?.(e);
+      }
+    };
+
+    useBackdropClick(closeMenu, [searchBoxRef, menuRef], isOpen);
+
+    return (
+      <LeafyGreenProvider darkMode={darkMode}>
+        <SearchInputContextProvider
+          state={state}
+          highlight={highlightIndex}
+          resultDynamicRefs={resultRefs}
+        >
+          <form
+            role="search"
+            ref={formRef}
+            className={cx(formStyle, className)}
+            onSubmit={handleSubmit}
+            onBlur={handleBlur}
+            {...rest}
+          >
+            {/* Disable eslint: onClick sets focus. Key events would already have focus */}
+            {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-element-interactions */}
+            <div
+              ref={searchBoxRef}
+              role="searchbox"
+              tabIndex={-1}
+              onMouseDown={handleSearchBoxMousedown}
+              onClick={handleSearchBoxClick}
+              onFocus={handleSearchBoxFocus}
+              onKeyDown={handleKeyDown}
+              className={cx(
+                inputWrapperStyle,
+                inputWrapperSizeStyle[sizeVariant],
+                inputWrapperThemeStyle[theme],
+                {
+                  [inputWrapperFocusStyles[theme]]:
+                    focusedElement === inputRef.current,
+                  [inputWrapperDisabledStyle[theme]]: disabled,
+                  [inputWrapperInteractiveStyles[theme]]: !disabled,
+                },
+              )}
+              aria-label={rest['aria-label']}
+              aria-labelledby={rest['aria-labelledby']}
+            >
+              <MagnifyingGlass
+                className={cx(
+                  searchIconStyle,
+                  searchIconThemeStyle[theme],
+                  searchIconSizeStyle[sizeVariant],
+                  { [searchIconDisabledStyle[theme]]: disabled },
+                )}
+                aria-label="Search Icon"
```

Comment:
nit are aria-label and role="presentation" oxy moronical? as in, i don't think a screen reader will announce the icon if it has role="presentation" set

Reply:
That's a good point. Technically `role` only removes the semantic meaning, but when used on an `svg` it will completely ignore it. rming label